treeherder/model/management/commands/cycle_data.py

```diff
@@ -7,17 +7,112 @@
                                      JobType,
                                      Machine,
                                      Repository)
-from treeherder.perf.models import PerformanceDatum
+from treeherder.perf.models import (PerformanceDatum,
+                                    PerformanceDatumManager)
+
+TREEHERDER = 'treeherder'
+PERFHERDER = 'perfherder'
+DATA_PRODUCERS = [TREEHERDER, PERFHERDER]
+
+
+class DataCycler:
+    source = ''
+
+    def __init__(self, days, chunk_size, sleep_time, is_debug=None,
+                 logger=None, **kwargs):
+        self.cycle_interval = datetime.timedelta(days=days)
+        self.chunk_size = chunk_size
+        self.sleep_time = sleep_time
+        self.is_debug = is_debug or False
+        self.logger = logger
+
+    def cycle(self, repository):
+        pass
+
+
+class TreeherderCycler(DataCycler):
+    source = TREEHERDER.title()
+
+    def cycle(self):
+        for repository in Repository.objects.all():
+            self.logger.debug("Cycling repository: {0}".format(repository.name))
```

Comment:
I've run the command manually and realized that having this use `logger.warning` would be really helpful since it prints to console.

Reply:
Got it. I'm updating the code to include it.